### PR TITLE
Adds sandbags

### PR DESCRIPTION
**Pull request by Bawhoppen**
Adds sandbags, these act just like sec barriers for cover from gunfire, however they aren't as good as at blocking bullets as sec barriers are.
They can be climbed over, and are table craftable from an empty sandbag and 5 sand, however due to the mapfreeze you can't get them ingame yet
ERT engis get some aswell.
They also have place-holderish sprites.

Code-wise tableclimbing has been moved to structures and made into a var

### CODE REVIEW

**Comment by Bawhoppen** [comment]
god fucking dammit I waited too long

File: code/game/objects/items/stacks/sheets/mineral.dm [hunk 1]
```diff
@@ -51,6 +52,32 @@ var/global/list/datum/stack_recipe/sandstone_recipes = list ( \
 	amount = 30
 
 /*
+ * Sandbags(
+ */
+
+/obj/item/stack/sheet/mineral/sandbags
```
Comment (KorPhaeron) [inline code/game/objects/items/stacks/sheets/mineral.dm]: >sheet/mineral

u wot

Comment (Jordie0608) [inline code/game/objects/items/stacks/sheets/mineral.dm]: Makes more sense than `/obj/item/stack/sheet/mineral/snow`

**Comment by KazeEspada** [comment]
Climb over atmos machinery.

**Comment by Bawhoppen** [comment]
wait no its not fixed something wrong

**Comment by Bawhoppen** [comment]
yes no i did indeed fuck something but i will manually fix it

**Comment by Bawhoppen** [comment]
okay now we can assume everything doesn't ruin everything else

**Comment by Bawhoppen** [comment]
example photo:
ss 2016-03-30 at 10 18 23

**Comment by Lumipharon** [comment]
Baw.
Baw~.

Don't miss this opportunity, make sand bags generic sack/hessian sack (also make it makable in the biogen, since that makes sense), an allow people to use empty sacks on people's heads, like cuffs, to blind them until removed.

Then I can finally be shitcurity/blackbagger.

**Comment by Bawhoppen** [comment]
@Lumipharon you're lucky you're getting this there was far too much shit gone wrong for this to be worth it but I struggled through

**Comment by Nienhaus** [comment]
Someone please make better sandbag sprites than mine.

**Comment by Bawhoppen** [comment]
oh nienhaus if you think your sprites are bad you should see the indivual item sandbags

**Comment by Lumipharon** [comment]
Well at the very least I would suggest making the bags constructable via biogen, not (from what I can tell) table crafting glass.

Fake edit: wait, that's making sandbags with glass, jesus christ how horrifying, a bag filled with broken glass shards would be a great ghetto weapon.

**Comment by Bawhoppen** [comment]
lumi thats ore/glass which is just shitty code for sand

**Comment by PKPenguin321** [comment]
Yeah, ore/glass is unrefined glass "ore," AKA sand
Source: I made pocket sand

**Comment by Lumipharon** [comment]
Fucking code, how does it work.

Either way though, bags producable out of the biogen, should be a 2 line change or somethin (or maybe not because nanoui?).

**Comment by Nienhaus** [comment]
@Bawhoppen That's a ripped image if I ever seen one.

**Comment by Bawhoppen** [comment]
@Nienhaus I did say it was placeholder

File: code/game/machinery/deployable.dm [hunk 2]
```diff
@@ -80,6 +80,10 @@
 /obj/structure/barricade/CanPass(atom/movable/mover, turf/target, height=0)//So bullets will fly over and stuff.
 	if(height==0)
 		return 1
+	if(density == 0)
```
Comment (phil235) [inline code/game/machinery/deployable.dm]: if(!density)

File: code/game/machinery/deployable.dm [hunk 3]
```diff
@@ -80,6 +80,10 @@
 /obj/structure/barricade/CanPass(atom/movable/mover, turf/target, height=0)//So bullets will fly over and stuff.
```
Comment (phil235) [inline code/game/machinery/deployable.dm]: it's probably tidier to do:
```
  	if(height==0)
 		return 1
	if(locate(/obj/structure/barricade) in get_turf(mover))
		return 1
	else if(istype(mover, /obj/item/projectile))
 		if(!anchored)
 			return 1
 		var/obj/item/projectile/proj = mover
 		if(proj.firer && Adjacent(proj.firer))
 			return 1
 		if(prob(proj_pass_rate))
 			return 1
 		return 0
	else
		return !density
```

**Comment by Bawhoppen** [comment]
Ok I did that thing you suggested phil, and they can now be made in the biogen for 200 bucks

**Comment by KorPhaeron** [comment]
Why not just let sand+backpack create sandbags?

Botany could still make them that way, but so could miners.

**Comment by Bawhoppen** [comment]
you're making me kill myself with that suggestion kor

File: code/game/objects/items/weapons/miscellaneous.dm [hunk 4]
```diff
@@ -17,4 +17,11 @@
 	icon = 'icons/obj/items.dmi'
 	icon_state = "skub"
 	w_class = 4
-	attack_verb = list("skubbed")
\ No newline at end of file
+	attack_verb = list("skubbed")
+
+/obj/item/weapon/emptysandbag
```
Comment (ChangelingRain) [inline code/game/objects/items/weapons/miscellaneous.dm]: I personally dislike having construction paths like this but eh

File: code/game/objects/structures.dm [hunk 5]
```diff
@@ -46,3 +49,66 @@
...
+		user.changeNext_move(CLICK_CD_MELEE)
+		structureclimber.Weaken(2)
+		structureclimber.visible_message("<span class='warning'>[structureclimber.name] has been knocked off the [src]", "You're knocked off the [src]!", "You see [structureclimber.name] get knocked off the [src]</span>")
+
+
+/obj/structure/CanPass(atom/movable/mover, turf/target, height=0)
+	if(height==0)
+		return 1
+	else
+		return !density
```
Comment (phil235) [inline code/game/objects/structures.dm]: You forgot to remove this after we discussed how to make your sandbag's climbing work. This broke roller bed.

File: code/game/objects/structures.dm [hunk 6]
```diff
@@ -46,3 +49,66 @@
...
+			else
+				user << "<span class='warning'>You fail to climb onto [src].</span>"
+			density = 1
+			structureclimber = null
+			return 1
+	structureclimber = null
+	return ..()
+
+
+/obj/structure/table/attack_hand(mob/living/user)
```
Comment (phil235) [inline code/game/objects/structures.dm]: didn't you mean /obj/structure/attack_hand()? why would you put table code in structure.dm?

Comment (Bawhoppen) [inline code/game/objects/structures.dm]: well I originally copied it over, I'll fix it soon

**Pull request merged**